test/test_value_ranges.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# Owner(s): ["oncall: pt2"]
+
+import itertools
+import math
+
+import sympy
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+    skipIfNoSympy,
+    TestCase,
+)
+from torch.utils._sympy.value_ranges import ValueRangeAnalysis, ValueRanges
+
+
+UNARY_OPS = [
+    "reciprocal",
+    "square",
+    "abs",
+    "neg",
+    "exp",
+    "log",
+    "sqrt",
+    "floor",
+    "ceil",
+]
+BINARY_OPS = ["truediv", "div", "add", "mul", "sub", "pow", "minimum", "maximum"]
+# a mix of constants, powers of two, primes
+CONSTANTS = [
+    -1,
+    0,
+    1,
+    2,
+    3,
+    4,
+    5,
+    8,
+    16,
+    32,
+    64,
+    100,
+    101,
+    2**24,
+    2**32,
+    2**37 - 1,
+]
+
+
+# The normal Python interpretation of the operators
+# TODO: maybe make this work with sympy?
+class ReferenceAnalysis:
+    @staticmethod
+    def reciprocal(x):
+        return 1 / x
+
+    @staticmethod
+    def square(x):
+        return x * x
+
+    @staticmethod
+    def abs(x):
+        return abs(x)
+
+    @staticmethod
+    def neg(x):
+        return -x
+
+    @staticmethod
+    def truediv(a, b):
+        return a / b
+
+    @staticmethod
+    def div(a, b):
+        return a // b
+
+    @staticmethod
+    def add(a, b):
+        return a + b
+
+    @staticmethod
+    def mul(a, b):
+        return a * b
+
+    @staticmethod
+    def sub(a, b):
+        return a - b
+
+    @staticmethod
+    def exp(x):
+        return sympy.exp(x)
```

Comment:
why use sympy instead of math ?

Reply:
Right now x might not be a sympy expression, and I need it to get sympified. This will become less important once I make the invariant that internally it's only sympy.Expr.